src/LaunchDarkly.EventSource/RetryDelayStrategy.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+namespace LaunchDarkly.EventSource
+{
+    /// <summary>
+    /// An abstraction of how <see cref="EventSource"/> should determine the delay
+    /// between retry attempts when a stream fails.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// The default behavior, provided by {@link DefaultRetryDelayStrategy}, provides
+    /// customizable exponential backoff and jitter. Applications may also create their own
+    /// implementations of RetryDelayStrategy if they desire different behavior. It is
+    /// generally a best practice to use backoff and jitter, to avoid a reconnect storm
+    /// during a service interruption.
+    /// </para>
+    /// <para>
+    /// Subclasses should be immutable. To implement strategies where the delay uses
+    /// different parameters on each subsequent retry (such as exponential backoff),
+    /// the strategy should return a new instance of its own class in
+    /// <see cref="Result.Next"/>, rather than modifying the state of the existing
+    /// instance. This makes it easy for EventSource to reset to the original delay
+    /// state when appropriate by simply reusing the original instance.
+    /// </para>
+    /// </remarks>
+    public abstract class RetryDelayStrategy
```

Comment:
Equivalent to the [Java version](https://github.com/launchdarkly/okhttp-eventsource/blob/main/src/main/java/com/launchdarkly/eventsource/RetryDelayStrategy.java).